tests/conftest.py

```diff
@@ -45,21 +53,25 @@
 from sample_data.data_org_type import data_org_type
 from sample_data.data_patch import data_patch
 from sample_data.data_population import data_population
-from sample_data.data_population_group import data_population_group
-from sample_data.data_population_status import data_population_status
 from sample_data.data_resource import data_resource
 from sample_data.data_sector import data_sector
 
 
 @pytest.fixture(scope="session")
-def engine():
-    engine = create_engine(url="sqlite:///:memory:")
-
-    # Execute pragma statement to enable foreign key constraints
-    with engine.connect() as conn:
-        conn.execute(text("PRAGMA foreign_keys = ON;"))
+def session():
+    engine = create_engine(
+        url="postgresql+psycopg://postgres:postgres@localhost:5432/hapitest"
+    )
+
+    # Create an empty schema
+    with engine.connect() as connection:
+        connection.execute(DropSchema("public", cascade=True, if_exists=True))
+        connection.commit()
+        connection.execute(CreateSchema("public", if_not_exists=True))
+        connection.commit()
 
     # Build the DB
+    build_hapi_views()
```

Comment:
Now I'm confused (about what I wrote :D), where were the views getting built before? Also why does this have to come before `create_all`?

Reply:
I think your code created the view on demand when running the test. I ran into some issue with it when switching to Postgres for testing, but forget now what it was after all the changes!

The build view code from the SQLAlchemy website https://github.com/sqlalchemy/sqlalchemy/wiki/Views#sqlalchemy-14-20-version creates a listener that works on table create. (It also has a listener to remove the view when the table is dropped). I realise the function is misnamed and will rename it as well as the ones in HDX Python Database since views are prepared rather than created.

Reply:
I've put the complexity into HDX Python Database

Reply:
Thanks for the explanation!